core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts

```diff
@@ -0,0 +1,690 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for exploration editor page component.
+ */
+
+import { TestBed } from '@angular/core/testing';
+import { ContextService } from 'services/context.service';
+import { EditabilityService } from 'services/editability.service';
+import { ExplorationFeaturesBackendApiService } from
+  'services/exploration-features-backend-api.service';
+import { ExplorationFeaturesService } from
+  'services/exploration-features.service';
+import { PageTitleService } from 'services/page-title.service';
+import { LoaderService } from 'services/loader.service';
+import { ParamChangesObjectFactory } from
+  'domain/exploration/ParamChangesObjectFactory';
+import { ParamSpecsObjectFactory } from
+  'domain/exploration/ParamSpecsObjectFactory';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { StateClassifierMappingService } from
+  'pages/exploration-player-page/services/state-classifier-mapping.service';
+import { StateEditorService } from
+  // eslint-disable-next-line max-len
+  'components/state-editor/state-editor-properties-services/state-editor.service';
+import { StateTopAnswersStatsBackendApiService } from
+  'services/state-top-answers-stats-backend-api.service';
+import { UserExplorationPermissionsService } from
+  'pages/exploration-editor-page/services/user-exploration-permissions.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+require('pages/exploration-editor-page/exploration-editor-page.component.ts');
+
+describe('Exploration editor page component', function() {
+  var ctrl = null;
+  var $q = null;
+  var $rootScope = null;
+  var $scope = null;
+  var $timeout = null;
+  var $uibModal = null;
+  var aims = null;
+  var cls = null;
+  var cs = null;
+  var es = null;
+  var efbas = null;
+  var efs = null;
+  var ers = null;
+  var ets = null;
+  var ews = null;
+  var gds = null;
+  var pts = null;
+  var rs = null;
+  var ses = null;
+  var stasbas = null;
+  var sas = null;
+  var tds = null;
+  var ueps = null;
+
+  var explorationId = 'exp1';
+  var explorationData = {
+    exploration_is_linked_to_story: true,
+    states: {
+      Introduction: {
+        param_changes: [],
+        content: {
+          html: '',
+        },
+        unresolved_answers: {},
+        interaction: {
+          customization_args: {},
+          answer_groups: [],
+          default_outcome: {
+            param_changes: [],
+            dest: 'Final',
+            feedback: {
+              html: '',
+              audio_translations: {}
```

Comment:
use content_id wherever you ate using html (As they are part of subtitledHtml object)

Reply:
Done.